src/main/scala/gestalt/Trees.scala

```diff
@@ -163,11 +162,19 @@ trait Trees extends Params with TypeParams with
   val TypeIdent: TypeIdentImpl
   trait TypeIdentImpl {
     def apply(name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[String]
   }
 
   val TypeSelect: TypeSelectImpl
   trait TypeSelectImpl {
     def apply(qual: Tree, name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[(Tree, String)]
+  }
+
+  val PathType: PathTypeImpl
+  trait PathTypeImpl {
+    def apply(qual: Option[Tree], name: String, targs: Seq[TypeTree]): TypeTree
+    def unapply(tpe: TypeTree) : Option[(Option[Tree], String, Seq[TypeTree])]
```

Comment:
BTW, if you encounter any difficulties, feel free to create a list of APIs that should be provided. I can work on them and ask people in the lab.